Source/Plugins/Core/com.equella.serverbase/src/com/tle/core/replicatedcache/ReplicatedCacheService.java

```
@@ -45,12 +46,41 @@
   <V extends Serializable> ReplicatedCache<V> getCache(
       @NonNull String cacheId, long maxLocalCacheSize, long ttl, @NonNull TimeUnit ttlUnit);
 
+  /**
+   * Similar to {@link #getCache(String, long, long, TimeUnit)}, but also allow persisting cache
+   * values in DB.
+   *
+   * @param cacheId unique identifier for this replicated cache.
+   * @param maxLocalCacheSize maximum number of values to be cache in-memory.
+   * @param ttl Time-to-live for cache values.
+   * @param ttlUnit units for ttl value.
+   * @param alwaysPersist `true` to persist cache values in Database.
+   */
+  <V extends Serializable> ReplicatedCache<V> getCache(
+      @NonNull String cacheId,
+      long maxLocalCacheSize,
+      long ttl,
+      @NonNull TimeUnit ttlUnit,
+      boolean alwaysPersist);
+
   @NonNullByDefault
-  public interface ReplicatedCache<V extends Serializable> {
+  interface ReplicatedCache<V extends Serializable> {
     Optional<V> get(@NonNull String key);
 
     void put(@NonNull String key, @NonNull V value);
 
+    /**
+     * Similar to { @link #put(String, Serializable)} }, but also allow specifying TTL of the value
+     * saved in DB.
+     *
+     * @param key Key of the object
+     * @param value Value of the object
+     * @param dbEntryTTL TTL of the value saved in DB
+     */
+    default void put(@NonNull String key, @NonNull V value, Date dbEntryTTL) {
```

Hmmm, I'm unsure about the default implementation simply ignoring the additional parameter. Feels like that could accidentally get very confusing. Would it be better to throw a `UnsupportedOperationException`?

And also:
* Is that name too specific, should it just be `ttl`?
* And why the old `Date` rather than `Instant` or one of the newer Java 8 date/time options?

---

The reason for `Date` is just because lower levels use `Date`, but I can definitely change the `Date` here to `Instant`.